programs/uxd/tests/integration_tests/cases/test_redeem_from_identity_depository.rs

```diff
@@ -0,0 +1,110 @@
+    program_test_context: &mut ProgramTestContext,
+    program_keys: &program_uxd::accounts::ProgramKeys,
+    payer: &Keypair,
+    user: &Keypair,
+    user_collateral: &Pubkey,
+    user_redeemable: &Pubkey,
+    redeemable_amount: u64,
+) -> Result<(), String> {
+    // Read state before
+    let redeemable_mint_before =
+        program_spl::accounts::read_token_mint(program_test_context, &program_keys.redeemable_mint)
+            .await?;
+    let controller_before =
+        program_uxd::accounts::read_controller(program_test_context, &program_keys.controller)
+            .await?;
+    let identity_depository_before = program_uxd::accounts::read_identity_depository(
+        program_test_context,
+        &program_keys.identity_depository_keys.depository,
+    )
+    .await?;
+
+    let redeemable_mint_supply_before = redeemable_mint_before.supply;
+    let redeemable_circulating_supply_before =
+        u64::try_from(controller_before.redeemable_circulating_supply).unwrap();
+    let redeemable_amount_under_management_before =
+        u64::try_from(identity_depository_before.redeemable_amount_under_management).unwrap();
+
+    let user_collateral_amount_before =
+        program_spl::accounts::read_token_account(program_test_context, user_collateral)
+            .await?
+            .amount;
+    let user_redeemable_amount_before =
+        program_spl::accounts::read_token_account(program_test_context, user_redeemable)
+            .await?
+            .amount;
+
+    // Execute
+    program_uxd::instructions::process_redeem_from_identity_depository(
+        program_test_context,
+        program_keys,
+        payer,
+        user,
+        user_collateral,
+        user_redeemable,
+        redeemable_amount,
+    )
+    .await?;
+
+    // Read state after
+    let redeemable_mint_after =
+        program_spl::accounts::read_token_mint(program_test_context, &program_keys.redeemable_mint)
+            .await?;
+    let controller_after =
+        program_uxd::accounts::read_controller(program_test_context, &program_keys.controller)
+            .await?;
+    let identity_depository_after = program_uxd::accounts::read_identity_depository(
+        program_test_context,
+        &program_keys.identity_depository_keys.depository,
+    )
+    .await?;
+
+    let redeemable_mint_supply_after = redeemable_mint_after.supply;
+    let redeemable_circulating_supply_after =
+        u64::try_from(controller_after.redeemable_circulating_supply).unwrap();
+    let redeemable_amount_under_management_after =
+        u64::try_from(identity_depository_after.redeemable_amount_under_management).unwrap();
+
+    let user_collateral_amount_after =
+        program_spl::accounts::read_token_account(program_test_context, user_collateral)
+            .await?
+            .amount;
+    let user_redeemable_amount_after =
+        program_spl::accounts::read_token_account(program_test_context, user_redeemable)
```

Comment:
Looks like `read_token_account` is exclusively used to read token account balance, would create utility function `get_token_account_balance(...)`

Reply:
Yeah was thinking about this too, there is no other interesting flag in the token account? I was trying to think about possible other usecases where we would read the token account for some other information. Maybe im overthinking this tho

Reply:
Actually I used the same idea than your other comment and created a type-generic functions for reading packed account instead so this function no longer exist :)